pkg/kn/commands/source/ping/create.go

```diff
@@ -70,9 +76,29 @@ func NewPingCreateCommand(p *commands.KnParams) *cobra.Command {
 			}
 			ceOverridesToRemove := util.ParseMinusSuffix(ceOverridesMap)
 
+			var data, dataBase64 string
+
+			switch updateFlags.encoding {
```

Comment:
IMO the same handling of `--encoding` flag should be added to `source ping update` as well. Probably extracting the code to `flags.go` to be shared.

Reply:
The `update` op should be able to handle switching between `data` and `dataBase64` field to clear the old one and change to the user's specified.